### PR TITLE
TeacherPostAcceptane Test

### PR DESCRIPTION
**Pull request by tobidub**

### CODE REVIEW

File: OtripleS.Web.Api.Tests.Acceptance/Brokers/OtripleSApiBroker.Teacher.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+using System.Threading.Tasks;
+using OtripleS.Web.Api.Models.Teachers;
+
+namespace OtripleS.Web.Api.Tests.Acceptance.Brokers
+{
+    public partial class OtripleSApiBroker
+    {
+        private const string TeachersRelativeUrl = "api/teachers";
+
+        public async ValueTask<Teacher> PostTeacherAsync(Teacher student) =>
```
Comment (HatemGamal) [inline OtripleS.Web.Api.Tests.Acceptance/Brokers/OtripleSApiBroker.Teacher.cs:12]: You need to rename parameters to be teacher instead of student

Comment (tobidub) [reply OtripleS.Web.Api.Tests.Acceptance/Brokers/OtripleSApiBroker.Teacher.cs]: good catch done

File: OtripleS.Web.Api.Tests.Acceptance/Brokers/OtripleSApiBroker.Teacher.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+        public async ValueTask<Teacher> DeleteTeacherByIdAsync(Guid teacherId) =>
+            await this.apiFactoryClient.DeleteContentAsync<Teacher>($"{TeachersRelativeUrl}/{teacherId}");
+
+        public async ValueTask<Teacher> PutTeacherAsync(Teacher teacher) =>
+            await this.apiFactoryClient.PutContentAsync(TeachersRelativeUrl, teacher);
+
+        public async ValueTask<List<Teacher>> GetAllTeachers() =>
+            await this.apiFactoryClient.GetContentAsync<List<Teacher>>($"{TeachersRelativeUrl}/");
+
```
Comment (hassanhabib) [inline OtripleS.Web.Api.Tests.Acceptance/Brokers/OtripleSApiBroker.Teacher.cs:26]: Remove extra empty line

File: OtripleS.Web.Api.Tests.Acceptance/Brokers/OtripleSApiBroker.Teacher.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
+﻿using System;
```
Comment (hassanhabib) [inline OtripleS.Web.Api.Tests.Acceptance/Brokers/OtripleSApiBroker.Teacher.cs:1]: Add copyrights notice

File: OtripleS.Web.Api.Tests.Acceptance/APIs/Teachers/TeachersApiTests.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
+﻿using System;
```
Comment (hassanhabib) [inline OtripleS.Web.Api.Tests.Acceptance/APIs/Teachers/TeachersApiTests.cs:1]: Add copyrights

File: OtripleS.Web.Api.Tests.Acceptance/APIs/Teachers/TeachersApiTests.Logic.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
+﻿using System.Threading.Tasks;
```
Comment (hassanhabib) [inline OtripleS.Web.Api.Tests.Acceptance/APIs/Teachers/TeachersApiTests.Logic.cs:1]: Add copyrights

File: OtripleS.Web.Api.Tests.Acceptance/APIs/Teachers/TeachersApiTests.Logic.cs [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+			Teacher actualTeacher =
+				await this.otripleSApiBroker.GetTeacherByIdAsync(inputTeacher.Id);
+
+			// then
+			actualTeacher.Should().BeEquivalentTo(expectedTeacher);
+
+			await this.otripleSApiBroker.DeleteTeacherByIdAsync(actualTeacher.Id);
+		}
+
```
Comment (hassanhabib) [inline OtripleS.Web.Api.Tests.Acceptance/APIs/Teachers/TeachersApiTests.Logic.cs:29]: Remove extra line

File: OtripleS.Web.Api.Tests.Acceptance/APIs/Teachers/TeachersApiTests.Logic.cs [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+			// when 
+			await this.otripleSApiBroker.PostTeacherAsync(inputTeacher);
+
+			Teacher actualTeacher =
+				await this.otripleSApiBroker.GetTeacherByIdAsync(inputTeacher.Id);
+
+			// then
+			actualTeacher.Should().BeEquivalentTo(expectedTeacher);
+
```
Comment (hassanhabib) [inline OtripleS.Web.Api.Tests.Acceptance/APIs/Teachers/TeachersApiTests.Logic.cs:26]: Remove extra line

**Pull request merged**